models/groups/index.ts

```diff
@@ -0,0 +1,77 @@
+import droleClient from '#clients/api-d-roles';
+import {
+  IDRolesGroup,
+  IDRolesGroupSearchResponse,
+} from '#clients/api-d-roles/interface';
+import { IAgentScope } from '#models/authentication/agent/scopes/constants';
+import { parseAgentScope } from '#models/authentication/agent/scopes/parse';
+import { FetchRessourceException } from '#models/exceptions';
+import { logFatalErrorInSentry } from '#utils/sentry';
+
+export class Groups {
+  /**
+   * Get all groups that a user belongs to
+   */
+  static async find(email: string): Promise<IDRolesGroupSearchResponse> {
+    try {
+      return await droleClient.getGroupsByEmail(email);
+    } catch (error) {
+      logFatalErrorInSentry(
+        new FetchRessourceException({
+          ressource: 'D-Roles Groups',
+          cause: error,
+        })
+      );
+      return [];
```

Comment:
Thinking. Should we return an empty list or completly fail the process. An empty would be very frustrating for a user that knows he belongs to some groups

Reply:
For login, it will result in a login failure.
Also, for action that requires admin rights, it will also fail.

I both case i think this is the wanted behaviour, i've made the change